lib/managers/files.js

```diff
@@ -1243,6 +1251,61 @@ Files.prototype.getCollaborations = function(fileID, options, callback) {
 };
 
 /**
+ * Enum of valid x-rep- hint values for generating representation info
+ *
+ * @readonly
+ * @enum {FileRepresentationType}
+ */
+Files.prototype.representation = {
+	PDF: '[pdf]',
+	THUMBNAIL: '[jpg?dimensions=320x320]',
+	IMAGE_MEDIUM: '[jpg?dimensions=1024x1024][png?dimensions=1024x1024]',
+	IMAGE_LARGE: '[jpg?dimensions=2048x2048][png?dimensions=2048x2048]',
+	EXTRACTED_TEXT: '[extracted_text]'
+};
+/**
+ * Requests information for all representation objects generated for a specific Box file
+ *
+ * API Endpoint: '/files/:fileID?fields=representations'
+ * Method : GET
+ *
+ * @param {string} fileID - Box ID of the file being requested
+ * @param {FileRepresentationType} representationType - The x-rep-hints value the application should create a
+ *    representation for. This value can either come from FileRepresentationType enum or manually created
+ * @param {Function} [callback] - A representation object containing the url_template, info, properties, and status
+ * @returns {Promise<Object>} A promise resolving to the representation response objects
+*/
+Files.prototype.getRepresentationInfo = function(fileID, representationType, callback) {
+	var params = {
+		qs: {
+			fields: 'representations'
+		},
+		headers: {
+			'x-rep-hints': representationType
+		}
+	};
+	var apiPath = urlPath(BASE_PATH, fileID);
+
+	return this.client.get(apiPath, params)
+		.then(response => {
+			switch (response.statusCode) {
+			// 202 - A Box file representation will be generated, but is not ready yet
+			case httpStatusCodes.ACCEPTED:
+				throw errors.buildResponseError(response, 'Representation not ready at this time');
+
+			// 200 - A Boxfile representation generated
+			// return the representation object
+			case httpStatusCodes.OK:
+				return response.body.representations;
```

Comment:
I just remembered that this return value is a collection it probably needs a case to handle if the user has the `iterators` option turned on!  See the [default response handler](https://github.com/box/box-node-sdk/blob/master/lib/box-client.js#L661) for an example of what we're doing, we need to do something similar here since.

Reply:
@mattwiller I took a iterators check from box-clients, let me know if that looks right!